micro_dl/utils/meta_utils.py

```diff
@@ -3,15 +3,89 @@
 import numpy as np
 import pandas as pd
 import sys
-import zarr.hierarchy
 
-import micro_dl.utils.aux_utils as aux_utils
-import micro_dl.utils.image_utils as im_utils
-import micro_dl.utils.io_utils as io_utils
 import micro_dl.utils.mp_utils as mp_utils
 from micro_dl.utils.cli_utils import show_progress_bar
 
 
+def write_meta_field(position: ngff.Position, metadata, field_name, subfield_name):
+    """
+    Writes 'metadata' to position's plate-level or FOV level .zattrs metadata by either
+    creating a new field (field_name) according to 'metadata', or updating the metadata
+    to an existing field if found, or concatenating the metadata from different channels.
+
+    Assumes that the zarr store group given follows the OMG-NGFF HCS
+    format as specified here:
+            https://ngff.openmicroscopy.org/latest/#hcs-layout
+
+    Warning: Dangerous. Writing metadata fields above the image-level of
+            an HCS hierarchy can break HCS compatibility
+
+    :param Position zarr_dir: NGFF position node object
+    :param dict metadata: metadata dictionary to write to JSON .zattrs
+    :param str subfield_name: name of subfield inside the the main field (values for different channels)
+    """
+    if field_name in position.zattrs:
+        if subfield_name in position.zattrs[field_name]:
+            position.zattrs[field_name][subfield_name].update(metadata)
+        else:
+            D1 = position.zattrs[field_name]
+            field_metadata = {
+                subfield_name: metadata,
+            }
+            # position.zattrs[field_name][subfield_name] = metadata
+            position.zattrs[field_name] = {**D1, **field_metadata}
+    else:
+        field_metadata = {
+            subfield_name: metadata,
+        }
+        position.zattrs[field_name] = field_metadata
+
+
+def add_channel(
```

Comment:
This function is only used once in `mp_utils.py`. I think it can be moved there instead.

Reply:
Makes sense! I have moved it to `mp_utils.py`. Thanks!